### PR TITLE
feat(artifact): add artifact component

### PR DESCRIPTION
**Pull request by chuang8511**
Because

- we want vdp able to access to artifact to enrich the capability of the products in instill AI

This commit

- add the schema of artifact component

### CODE REVIEW

**Comment by chuang8511** [comment]
Hi @donch1989 ,
According to the new flow, you can review schema first.

**Review by donch1989 — CHANGES_REQUESTED** [review]

File: data/artifact/v0/config/setup.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      ],
+      "instillAcceptFormats": [
+        "string"
+      ],
+      "instillSecret": true,
+      "instillUIOrder": 0,
+      "title": "API Token",
+      "type": "string"
+    },
+    "namespace": {
```
[the remark this answers is not in the document]
Comment (chuang8511) [reply data/artifact/v0/config/setup.json:18]: @donch1989 
I think namespace is more like "account name" such as login information, so I put it in setup section.
Will it be better to put them in tasks section?

**Comment by chuang8511** [comment]
todo: take out api token

**Comment by chuang8511** [comment]
@donch1989 
I have
- moved namespace to tasks
- take out api token

**Review by donch1989 — APPROVED** [review]
LGTM

**Pull request merged**